### PR TITLE
Fixes and improvements to Bittrex WS

### PR DESCRIPTION
**Pull request by CjS77**
This also fixes the WS sample for bittrex api.

### CODE REVIEW

File: src/exchanges/bittrex/BittrexFeed.ts [hunk 1]
```diff
@@ -46,38 +46,60 @@ export class BittrexFeed extends ExchangeFeed {
         return 'Bittrex';
     }
 
-    subscribe(products: string[]): boolean {
+    subscribe(products: string[]): Promise<boolean> {
```
Comment (CjS77) [inline src/exchanges/bittrex/BittrexFeed.ts]: If we make the `subscribe` function an "all-or-nothing" affair, then we can guarantee that the array of promises only resolve if they all return true. We can then skip the checks and simplify the `then` part of the logic.

File: src/exchanges/bittrex/BittrexFeed.ts [hunk 2]
```diff
@@ -46,38 +46,60 @@ export class BittrexFeed extends ExchangeFeed {
...
+                            return reject(new Error(msg));
+                        }
+                        const snapshot: SnapshotMessage = this.processSnapshot(product, data);
+                        this.push(snapshot);
+                        return resolve(true);
+                    });
+                });
             });
+        })).then(() => {
+            // Every result is guaranteed to be true
```
Comment (CjS77) [inline src/exchanges/bittrex/BittrexFeed.ts]: I'm leaning towards making all GTT promises reject if the result isn't 100% as expected. Not all classes have migrated to this approach yet, but I believe it's much safer to be conservative when money is involved.  

One can always add `catch` clauses in user code to handle unexpected cases.

File: src/factories/bittrexFactories.ts [hunk 3]
```diff
@@ -45,9 +45,12 @@ export function getSubscribedFeeds(options: ExchangeFeedConfig, products: string
             return reject(new Error('TIMEOUT. Could not connect to Bittrex Feed server'));
         }, 30000);
         feed.on('websocket-connection', () => {
-            feed.subscribe(products);
-            clearTimeout(timeout);
-            return resolve(feed);
+            feed.subscribe(products).then(() => {
```
Comment (CjS77) [inline src/factories/bittrexFactories.ts]: Since we return a promise now, may as well use it properly :)

File: src/samples/bittrexWSdemo.ts [hunk 4]
```diff
@@ -26,36 +26,42 @@ const product: string = 'BTC-ETH';
 const config: ExchangeFeedConfig = {
     logger: logger,
     auth: null,
-    wsUrl: null,
+    wsUrl: null
 };
```
Comment (CjS77) [inline src/samples/bittrexWSdemo.ts]: Just rearranging things a bit to make use of the returned promise.

**Comment by CjS77** [comment]
@cjpetrus Thanks for this PR! I've made a few minor changes; if you're happy, I'll merge it in.

**Comment by themagicmountain** [comment]
wait wait! This is actually my commit! I just commented in issue #78 (and some other issue that's now been deleted?) with the repository, but @cjpetrus did the right thing by making a pull request. Your edits make sense to me, although now I'd like to advocate for not reducing the promises when returning from subscribe -- https://github.com/themagicmountain/gdax-tt/blob/bittrex/src/exchanges/bittrex/BittrexFeed.ts, and to have the "user" handle the flattening/reducing, as in https://github.com/themagicmountain/gdax-tt/blob/bittrex/src/factories/bittrexFactories.ts (since if i subscribe to 3 channels,I'd want to know which of the three failed, not that one of them did)

File: src/exchanges/bittrex/BittrexFeed.ts (hunk 2 above)
Comment (cjpetrus) [reply src/exchanges/bittrex/BittrexFeed.ts]: I feel the same way.Even though by nature, programmatic trading is unsafe. ¯\_(ツ)_/¯ best we can hope for is to minimize risk.

Comment (themagicmountain) [reply src/exchanges/bittrex/BittrexFeed.ts]: wait wait! This is actually my commit! I just commented in issue #78 (and some other issue that's now been deleted?) with the repository, but @cjpetrus did the right thing by making a pull request.  I think it should actually return the unmerge array -- https://github.com/themagicmountain/gdax-tt/blob/bittrex/src/exchanges/bittrex/BittrexFeed.ts, and to have the "user" handle the flattening/reducing, as in https://github.com/themagicmountain/gdax-tt/blob/bittrex/src/factories/bittrexFactories.ts (and ultimately choose to reject in the factory function?)

File: src/factories/bittrexFactories.ts (hunk 3 above)
Comment (cjpetrus) [reply src/factories/bittrexFactories.ts]: indeed. coming from the land of beautiful async await py, I tend to miss things like this.

**Pull request merged**